src/core/ddsc/include/dds/ddsc/dds_public_qos.h

```diff
@@ -1010,6 +1027,23 @@ dds_qget_prop (
   const char * name,
   char ** value);
 
+/**
```

Comment:
Here I would follow the same pattern as what I suggested for the setter: a `dds_qget_prop_propagate` that *also* returns the setting of the "propagate" flag (if a non-null pointer is passed in), then implement `dds_qget_prop` as:
```
bool dds_qget_prop (const dds_qos_t *qos, const char *name, char **value)
{
  return dds_qget_prop_propagate (qos, name, value, NULL);
}
```

(And again the same for the binary property one.)